engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/emsum/BigDecimalEMSOperator.java

```diff
@@ -0,0 +1,121 @@
+package io.deephaven.engine.table.impl.updateby.emsum;
+
+import io.deephaven.api.updateby.OperationControl;
+import io.deephaven.chunk.Chunk;
+import io.deephaven.chunk.LongChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.MatchPair;
+import io.deephaven.engine.table.impl.updateby.UpdateByOperator;
+import io.deephaven.engine.table.impl.util.RowRedirection;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigDecimal;
+
+import static io.deephaven.util.QueryConstants.NULL_LONG;
+
+public class BigDecimalEMSOperator extends BaseBigNumberEMSOperator<BigDecimal> {
+    public class Context extends BaseBigNumberEMSOperator<BigDecimal>.Context {
+        protected Context(final int chunkSize) {
+            super(chunkSize);
+        }
+
+        @Override
+        public void accumulateCumulative(RowSequence inputKeys,
+                Chunk<? extends Values>[] valueChunkArr,
+                LongChunk<? extends Values> tsChunk,
+                int len) {
+            setValuesChunk(valueChunkArr[0]);
+
+            // chunk processing
+            if (timestampColumnName == null) {
+                // compute with ticks
+                for (int ii = 0; ii < len; ii++) {
+                    // read the value from the values chunk
+                    final BigDecimal input = objectValueChunk.get(ii);
+                    if (input == null) {
+                        handleBadData(this, true);
+                    } else {
+                        if (curVal == null) {
```

Comment:
I would combine the `else` and `if` to decrease a layer of unnecessary indentation.